### PR TITLE
Add config for running extension tests

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Add .bat for running extension unit tests (not full integration tests, currently only Notebook) and set up launch.json with 2 new launch configs for running & debugging extension tests.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -147,7 +147,23 @@
...
+			"request": "launch",
+            "name": "Run Extension Unit Tests",
+			"windows": {
+				"runtimeExecutable": "${workspaceFolder}/scripts/test-extensions-unit.bat"
+			},
+			"osx": {
+				"runtimeExecutable": "${workspaceFolder}/scripts/test-extensions-unit.sh"
+			},
+			"linux": {
+				"runtimeExecutable": "${workspaceFolder}/scripts/test-extensions.sh"
```
Comment (MattIrv) [inline .vscode/launch.json]: why is this different for mac and linux? Usually they're the same

Edit: Below it looks like you renamed the file so I expect this just needs to be updated

File: .vscode/launch.json [hunk 2]
```diff
@@ -147,7 +147,23 @@
 			"outFiles": [
 				"${workspaceFolder}/out/**/*.js"
 			]
-		}
+        },
+        {
+			"type": "chrome",
+			"request": "launch",
+            "name": "Run Extension Unit Tests",
```
Comment (MattIrv) [inline .vscode/launch.json]: tabbing here and below my next comment is a bit off

File: .vscode/launch.json [hunk 3]
```diff
@@ -156,7 +172,14 @@
...
 			]
-		},
+        },
+        {
+            "name": "Debug Extension Unit Tests",
+            "configurations": [
+                "Attach to Extension Host",
+                "Run Extension Unit Tests"
+            ]
+        },
```
Comment (MattIrv) [inline .vscode/launch.json]: tabbing is off here for this block too

File: .vscode/launch.json [hunk 4]
```diff
@@ -147,7 +147,23 @@
...
+				"runtimeExecutable": "${workspaceFolder}/scripts/test-extensions-unit.bat"
+			},
+			"osx": {
+				"runtimeExecutable": "${workspaceFolder}/scripts/test-extensions-unit.sh"
+			},
+			"linux": {
+				"runtimeExecutable": "${workspaceFolder}/scripts/test-extensions-unit.sh"
+			},
+			"webRoot": "${workspaceFolder}",
+			"timeout": 45000
```
Comment (chlafreniere) [inline .vscode/launch.json]: I'm curious if this timeout is long enough

File: scripts/test-extensions-unit.bat [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+setlocal
+
+pushd %~dp0\..
+
+set VSCODEUSERDATADIR=%TMP%\adsuser-%RANDOM%-%TIME:~6,5%
+set VSCODEEXTENSIONSDIR=%TMP%\adsext-%RANDOM%-%TIME:~6,5%
+echo %VSCODEUSERDATADIR%
+echo %VSCODEEXTENSIONSDIR%
+@echo OFF
```
Comment (chlafreniere) [inline scripts/test-extensions-unit.bat]: do we need to turn "@ echo" back ON (I'm guessing no, right?)

**Pull request merged**